### PR TITLE
fix make_initial_point_fn

### LINKED ISSUE #60: After pymc update there is an issue with model compilation
AttributeError: module 'pymc.model' has no attribute 'make_initial_point_fn'

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[10], line 1
----> 1 compiled_model = nutpie.compile_pymc_model(model)

File ~/.cache/pypoetry/virtualenvs/***-RmO9ALKT-py3.10/lib/python3.10/site-packages/nutpie/compile_pymc.py:170, in compile_pymc_model(model, **kwargs)
    146 def compile_pymc_model(model: pm.Model, **kwargs) -> CompiledPyMCModel:
    147     """Compile necessary functions for sampling a pymc model.
    148 
    149     Parameters
   (...)
    158 
    159     """
    161     (
    162         n_dim,
    163         n_expanded,
    164         logp_fn_pt,
    165         logp_fn,
    166         expand_fn_pt,
    167         expand_fn,
    168         shared_expand,
    169         shape_info,
--> 170     ) = _make_functions(model)
    172     shared_data = {val.name: val.get_value().copy() for val in logp_fn_pt.get_shared()}
    173     for val in shared_data.values():

File ~/.cache/pypoetry/virtualenvs/***-RmO9ALKT-py3.10/lib/python3.10/site-packages/nutpie/compile_pymc.py:248, in _make_functions(model)
    247 def _make_functions(model):
--> 248     shapes = _compute_shapes(model)
    250     # Make logp_dlogp_function
    251     joined = pt.dvector("__joined_variables")

File ~/.cache/pypoetry/virtualenvs/***-RmO9ALKT-py3.10/lib/python3.10/site-packages/nutpie/compile_pymc.py:222, in _compute_shapes(model)
    221 def _compute_shapes(model):
--> 222     point = pm.model.make_initial_point_fn(model=model, return_transformed=True)(0)
    224     trace_vars = {
    225         var.name: var
    226         for var in model.value_vars + model.free_RVs + model.deterministics
    227         if var not in model.observed_RVs + model.potentials
    228     }
    230     shape_func = pytensor.compile.function.function(
    231         inputs=[],
    232         outputs=[var.shape for var in trace_vars.values()],
   (...)
    242         on_unused_input="ignore",
    243     )

AttributeError: module 'pymc.model' has no attribute 'make_initial_point_fn'
```

### PR DESCRIPTION
**Pull request by ferrine**
fix #60

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -30,10 +30,10 @@ dependencies = [
 
 [project.optional-dependencies]
 stan = ["bridgestan >= 2.1.2"]
-pymc = ["pymc >= 5.5.0", "numba >= 0.57.1"]
+pymc = ["pymc >= 5.8.0", "numba >= 0.57.1"]
 all = [
     "bridgestan >= 2.1.2",
-    "pymc >= 5.5.0",
+    "pymc >= 5.8.0",
```
Comment (ricardoV94) [inline pyproject.toml:36]: This isn't needed. The new path is the correct one for the older versions as well

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
Is this ready @ferrine?

File: pyproject.toml (hunk 1 above)
Comment (ferrine) [reply pyproject.toml]: reverted any changes to pymc versions

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "nutpie"
-version = "0.8.1"
+version = "0.9.1"
```
Comment (twiecki) [inline Cargo.toml:3]:
```suggestion
version = "0.9.0"
```

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
https://github.com/pymc-devs/nutpie/blob/main/CHANGELOG.md

**Comment by ricardoV94** [comment]
> https://github.com/pymc-devs/nutpie/blob/main/CHANGELOG.md

That's very cumbersome compared to automatic release notes (and I don't think it's been properly updated)

**Comment by twiecki** [comment]
@ricardoV94 then we can just merge

**Pull request merged**